packages/@react-spectrum/dnd/docs/dragAndDrop.mdx

```diff
@@ -0,0 +1,246 @@
+provide a `getDropOperation` to `useDropHooks` that returns a "cancel" drop operation if any of the item types includes "folder".
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show code</summary>
+
+```tsx example export=true render=false
+import {useDropHooks} from '@react-spectrum/dnd';
+
+function DroppableList(props) {
+  let {
+    items,
+    ...otherProps
+  } = props;
+  let list = useListData({
+    initialItems: items || [
+      {id: 'f', textValue: 'Adobe AfterEffects', type: 'file'},
+      {id: 'g', textValue: 'Adobe Illustrator', type: 'file'},
+      {id: 'h', textValue: 'Adobe Lightroom', type: 'file'},
+      {id: 'i', textValue: 'Adobe Premiere Pro', type: 'file'},
+      {id: 'j', textValue: 'Adobe Fresco', type: 'file'}
+    ]
+  });
+
+  let onDrop = async (e) => {
+    let itemsToAdd = [];
+    for (let item of e.items) {
+      let type, id, textValue;
+      if (item.kind === 'text') {
+        if (item.types.has('file')) {
+          type = 'file';
+          id = await item.getText('file');
+        }
+
+        if (item.types.has('text/plain')) {
+          textValue = await item.getText('text/plain');
+        }
+
+        if (!type || !id || !textValue) {
+          continue;
+        }
+
+        itemsToAdd.push({
+          id,
+          type,
+          textValue
+        });
+      }
+    }
+
+    if (e.target.type === 'root') {
+      list.prepend(...itemsToAdd);
+    } else if (e.target.dropPosition === 'before') {
+      list.insertBefore(e.target.key, ...itemsToAdd);
+    } else {
+      list.insertAfter(e.target.key, ...itemsToAdd);
+    }
+  };
+
+  let getDropOperation = (target, types) => {
+    if (types.has('folder')) {
+      return 'cancel';
+    }
+
+    return 'move';
+  };
+
+  let dropHooks = useDropHooks({
+    getDropOperation,
+    onDrop
+  });
+
+  return (
+    <ListView
+      aria-label="Droppable list view example"
+      width="size-3600"
+      height="size-3600"
+      selectionMode="multiple"
+      items={list.items}
+      dropHooks={dropHooks}
+      {...otherProps}>
+      {item => (
+        <Item key={item.key} textValue={item.textValue}>
+          {item.type === 'folder' && <Folder />}
+          <Text>{item.textValue}</Text>
+        </Item>
+      )}
+    </ListView>
+  );
+}
+```
+</details>
+
+To perform a drag and drop operations via a keyboard, first select the items to be dragged. You can then start the drag operation by focusing the drag handle on any of the
+selected rows and hitting <Keyboard>Enter</Keyboard>/<Keyboard>Space</Keyboard>. Once a drag operation is started, <Keyboard>Tab</Keyboard> can be used to cycle through valid
+drop targets. Hitting <Keyboard>Enter</Keyboard> will then confirm the drop operation on the focused drop target. For screen readers, please follow the instructions announced
+when focusing the row's drag handle to begin a drag operation. Go ahead and try out drag and drop in the example below!
```

Comment:
Including this description of how to drag and drop with the other interaction modes since it feels like it would be pretty common across our drag and drop compatible components

Reply:
If we put this at the top of the page, it probably won't need to be repeated here.

Reply:
Ok I'll put it in the intro section